    @Test
    public void bug51187() throws Exception {
       XMLSlideShow ss1 = XSLFTestDataSamples.openSampleDocument("51187.pptx");
       
       assertEquals(1, ss1.getSlides().size());
       
       // Check the relations on it
       // Note - rId3 is a self reference
       XSLFSlide slide0 = ss1.getSlides().get(0);

       assertEquals("/ppt/slides/slide1.xml", slide0.getPackagePart().getPartName().toString());
       assertEquals("/ppt/slideLayouts/slideLayout12.xml", slide0.getRelationById("rId1").getPackageRelationship().getTargetURI().toString());
       assertEquals("/ppt/notesSlides/notesSlide1.xml", slide0.getRelationById("rId2").getPackageRelationship().getTargetURI().toString());
       assertEquals("/ppt/slides/slide1.xml", slide0.getRelationById("rId3").getPackageRelationship().getTargetURI().toString());
       assertEquals("/ppt/media/image1.png", slide0.getRelationById("rId4").getPackageRelationship().getTargetURI().toString());
       
       // Save and re-load
       XMLSlideShow ss2 = XSLFTestDataSamples.writeOutAndReadBack(ss1);
       ss1.close();
       assertEquals(1, ss2.getSlides().size());
       
       slide0 = ss2.getSlides().get(0);
       assertEquals("/ppt/slides/slide1.xml", slide0.getPackagePart().getPartName().toString());
       assertEquals("/ppt/slideLayouts/slideLayout12.xml", slide0.getRelationById("rId1").getPackageRelationship().getTargetURI().toString());
       assertEquals("/ppt/notesSlides/notesSlide1.xml", slide0.getRelationById("rId2").getPackageRelationship().getTargetURI().toString());
       // TODO Fix this
       assertEquals("/ppt/slides/slide1.xml", slide0.getRelationById("rId3").getPackageRelationship().getTargetURI().toString());
       assertEquals("/ppt/media/image1.png", slide0.getRelationById("rId4").getPackageRelationship().getTargetURI().toString());
       
       ss2.close();
    }

